lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -321,30 +321,24 @@ export class PublicClientApplication implements IPublicClientApplication {
             // Create acquire token url.
             const navigateUrl = await authClient.getAuthCodeUrl(validRequest);
 
-            // Acquire token with popup
-            return await this.popupTokenHelper(navigateUrl, authCodeRequest, authClient);
+            // Create popup interaction handler.
+            const interactionHandler = new PopupHandler(authClient, this.browserStorage);
+
+            // Show the UI once the url has been created. Get the window handle for the popup.
+            const popupWindow: Window = interactionHandler.initiateAuthRequest(navigateUrl, authCodeRequest);
+
+            // Monitor the window for the hash. Return the string value and close the popup when the hash is received. Default timeout is 60 seconds.
+            const hash = await interactionHandler.monitorPopupForHash(popupWindow, this.config.system.windowHashTimeout);
+
+            // Handle response from hash string.
+            return await interactionHandler.handleCodeResponse(hash);
```

Comment:
The catch is a single catch for all issues seen in initialization, creating auth code client and handling the code response. Is that alright?

Reply:
Not sure what you mean? Since we cache items in the request preflighting, we need to make sure that we clear it if there are any errors that happen.

Reply:
+ cache the error for telemetry purposes. This is done in every API